src/components/molecules/FormFields/AssetSelection.tsx

```diff
@@ -82,6 +84,9 @@ export default function AssetSelection({
                   {...props}
                   disabled={disabled}
                   value={asset.did}
+                  onClick={(e: any) => {
```

Comment:
this can lead to unintended consequences. Like, it just keeps `asset.checked` as `true` without deselecting it again. But instead of replicating formik functionality we should rather figure out to get selection state from existing functionality. Like:

- figure out if we can use `asset.checked` for keeping checked state instead of only using it for the default selected state
- the `value` prop in `AssetSelection` always holds the DID of the currently selected asset. So figure out how to use this to figure out what user has selected

Reply:
Thanks, this is helpful 
I am thinking of a solution like this:
```
  const [selectedAlgorithm, setSelectedAlgorithm] = useState('')
  useEffect(() => {
    setSelectedAlgorithm(props.value)
  }, [props.value])
```
And then to display the status check:
```
  {selectedAlgorithm === asset.did && (
    <AssetStatus ddo={DID.parse(asset.did)} />
  )}
```